nodejs/awsx/cloudtrail/trail.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import * as utils from "../utils";
+
+export class Trail extends pulumi.ComponentResource {
+    /**
+     * The managed CloudWatch Log Group.
+     */
+    public readonly logGroup: aws.cloudwatch.LogGroup | undefined;
+
+    /**
+     * The managed S3 Bucket.
+     */
+    public readonly bucket: aws.s3.Bucket | undefined;
+
+    /**
+     * The CloudWatch Trail.
+     */
+    public readonly trail: aws.cloudtrail.Trail;
+
+    /**
+     * Create a Trail resource with the given unique name, arguments, and options.
+     *
+     * @param name The _unique_ name of the resource.
+     * @param args The arguments to use to populate this resource's properties.
+     * @param opts A bag of options that control this resource's behavior.
+     */
+    constructor(name: string, args: TrailArgs, opts?: pulumi.CustomResourceOptions) {
+        super("aws:cloudtrail:x:Trail", name, {}, opts);
+
+        let bucketName: pulumi.Output<string>;
+        if (args.s3BucketName !== undefined) {
+            bucketName = pulumi.output(args.s3BucketName);
+        } else {
+            this.bucket = bucketWithCloudTrailPolicy(name, this);
+            bucketName = this.bucket.id;
+        }
+
+        let cloudWatchLogGroupArn: pulumi.Output<string> | undefined = undefined;
+        if (args.sendToCloudWatchLogs) {
+            this.logGroup = new aws.cloudwatch.LogGroup(name, args.cloudWatchLogGroupArgs, {parent: this});
+            cloudWatchLogGroupArn = pulumi.interpolate`${this.logGroup.arn}:*`;
+        }
+
+        const trailArgs = {
+            s3BucketName: bucketName,
+            cloudWatchLogGroupArn: cloudWatchLogGroupArn,
+            ...args,
+        };
+
+        this.trail = new aws.cloudtrail.Trail(name, trailArgs, {parent: this});
+    }
+}
+
+
+type TrailArgs = utils.Overwrite<aws.cloudtrail.TrailArgs, {
+    /**
+     * If CloudTrail pushes logs to CloudWatch Logs in addition to S3.
+     *
+     * Disabled for cost out of the box.
+     *
+     * @default false
+     */
+    readonly sendToCloudWatchLogs?: pulumi.Input<boolean>;
```

Comment:
In the constructor above, we read `sendToCloudWatchLogs` as a prompt value, so this can't be an `Input`. I also wouldn't expect this or any of the arguments in this interface to be marked as `readonly`.

Reply:
I'll remove the `readonly` and `Input`, but I was duplicating what has been done in the existing interface: https://github.com/pulumi/pulumi-aws/blob/master/sdk/nodejs/cloudtrail/trail.ts#L390-L405

Reply:
Makes sense, but the rest of this package doesn't do that, so I think the consistency (of not having `readonly`) is preferable.